microsoft-edge/webview2/concepts/print.md

```diff
@@ -0,0 +1,736 @@
+  CHECK_FAILURE(printSettings2->put_ShouldPrintHeaderAndFooter(samplePrintSettings.HeaderAndFooter));
+  CHECK_FAILURE(printSettings2->put_HeaderTitle(samplePrintSettings.HeaderTitle.c_str()));
+  CHECK_FAILURE(printSettings2->put_FooterUri(samplePrintSettings.FooterUri.c_str()));
+  CHECK_FAILURE(printSettings2->put_PrinterName(printerName.c_str()));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(webView2_16->Print(
+      printSettings.get(),
+      Callback<ICoreWebView2PrintCompletedHandler>(
+          [title = std::move(title), this](HRESULT errorCode, COREWEBVIEW2_PRINT_STATUS printStatus) -> HRESULT
+          {
+            std::wstring message = L"";
+            if (errorCode == S_OK && printStatus == COREWEBVIEW2_PRINT_STATUS_SUCCEEDED)
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                              L" document to printer is succedded";
+            }
+            else if (errorCode == S_OK && printStatus == COREWEBVIEW2_PRINT_STATUS_PRINTER_UNAVAILABLE)
+            {
+              message = L"Selected printer is not found, not available, offline or "
+                              L"error state.";
+            }
+            else if (errorCode == E_INVALIDARG)
+            {
+              message = L"Invalid settings provided for the specified printer";
+            }
+            else if (errorCode == E_ABORT)
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                                  L" document already in progress";
+            }
+            else
+            {
+              message = L"Printing " + std::wstring(title.get()) +
+                                  L" document to printer is failed";
+            }
+
+            AsyncMessageBox(message, L"Print to printer");
+
+            return S_OK;
+          })
+          .Get()));
+  return true;
+}
+```
+
+**APIs:**
+
+* [ICoreWebView2_16::Print() method](/microsoft-edge/webview2/reference/win32/icorewebview2_16#print)
+* [ICoreWebView2PrintSettings interace](microsoft-edge/webview2/reference/win32/icorewebview2printsettings1)
+* [ICoreWebView2PrintSettings2 interace](microsoft-edge/webview2/reference/win32/icorewebview2printsettings2)
+* [COREWEBVIEW2_PRINT_STATUS Enum](/microsoft-edge/webview2/reference/win32/icorewebview2_16#corewebview2_print_status)
+
+---
+
+<!-- end of tab-set -->
+
+
+<!-- ====================================================================== -->
+## PrintToPdf
+
+Silently print the current top-level document in the WebView2 to a PDF file. To completely control how printing is performed, you can print to a PDF and then build your own code to print the PDF. This API consists of an asynchronous PrintToPdf method and a PrintSettings object. The PrintToPdf method accepts a path that the PDF file should be saved to.
```

Comment:
```suggestion
Silently prints the current top-level document in the WebView2 control to a PDF file. To completely control how printing is performed, you can print to a PDF and then build your own code to print the PDF. This API consists of an asynchronous `PrintToPdf` method and a `PrintSettings` object. The `PrintToPdf` method accepts a path that the PDF file will be saved to.
```
not always applicable:
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/should-vs-must